packages/components/src/public/QueryModel/TabbedGridPanel.tsx

```diff
@@ -114,6 +126,42 @@ export const TabbedGridPanel: FC<TabbedGridPanelProps & InjectedQueryModels> = m
         },
         [onTabSelect]
     );
+
+    const exportTabs = useCallback(async (selectedTabs: string[]|Set<string>):Promise<void> => {
+        try {
+            // set exporting blocker
+            setCanExport(false);
+            let models = [];
+            selectedTabs.forEach(selected => models.push(queryModels[selected]?.schemaQuery));
+            const filename = exportFilename ?? 'Data';
+            await exportTabsXlsx(filename, models);
+            onExport?.[EXPORT_TYPES.EXCEL]?.();
+            createNotification({message: 'Successfully exported tabs to file.', alertClass: 'success'});
+        } catch (e) {
+            //Set export error
+            createNotification({message: 'Export failed: ' + e.getErrorMessage(), alertClass: 'danger'});
+        } finally {
+            // unset exporting blocker
+            setShowExportModal(false);
+            setCanExport(true);
+        }
+    }, [exportFilename, canExport, queryModels]);
+
+    const exportHandler = useCallback(async () => {
+        if (Object.keys(tabOrder).length > 1) {
+            setShowExportModal(true);
+            return;
+        }
+
+        exportTabs([internalActiveId]).then();
```

Comment:
Since `exportTabs` uses async/await, I don't think the return type needs to be a `Promise`. If that return type is changed to just `void` then you should be able to remove the `then()` call here.

Reply:
removed, seems like it still wants the promise